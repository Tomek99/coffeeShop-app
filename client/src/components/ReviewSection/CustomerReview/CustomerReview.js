import React from "react";
import styles from "./CustomerReview.module.scss";
import PropTypes from "prop-types";
import RatingsStars from "../../RatingStars/RatingStars";

function CustomerReview({ item }) {
  const { id, imageUrl, text, avatarUrl, name, rate } = item;

  return (
    <div key={id} className={styles.singleCustomer}>
      <div className={styles.singleCustomerLeftDiv}>
        <img src={avatarUrl} alt="" className={styles.avatar} />
        <p className={styles.name}>{name}</p>
        <div>
          <RatingsStars rate={rate} size="medium" />
        </div>
      </div>
      <p>{text}</p>
    </div>
  );
}

CustomerReview.propTypes = {
  id: PropTypes.number,
  imageUrl: PropTypes.string,
  text: PropTypes.string,
  avatarUrl: PropTypes.string,
  name: PropTypes.string,
  rate: PropTypes.number,
};
export default CustomerReview;
