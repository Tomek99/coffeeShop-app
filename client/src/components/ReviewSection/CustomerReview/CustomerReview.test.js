import { render, screen } from "@testing-library/react";
import CustomerReview from "./CustomerReview";

const item = {
  id: "31232",
  imageUrl: "",
  text: " test test test",
  avatarUrl: "",
  name: "productName",
  rate: 5,
};
test("should render CustomerReview", () => {
  const { container } = render(<CustomerReview item={item} />);

  expect(container).toMatchSnapshot();
});
