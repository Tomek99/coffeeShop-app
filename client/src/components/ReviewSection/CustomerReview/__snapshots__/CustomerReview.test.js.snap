// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render CustomerReview 1`] = `
<div>
  <div
    class="singleCustomer"
  >
    <div
      class="singleCustomerLeftDiv"
    >
      <img
        alt=""
        class="avatar"
        src=""
      />
      <p
        class="name"
      >
        productName
      </p>
      <div>
        <span
          aria-label="5 Stars"
          class="MuiRating-root MuiRating-sizeMedium MuiRating-readyOnly RatingsStars css-ryrseu-MuiRating-root"
          role="img"
        >
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
          <span>
            <span
              class="MuiRating-icon MuiRating-iconFilled css-7qmtgc-MuiRating-icon"
            >
              <svg
                aria-hidden="true"
                class="MuiSvgIcon-root MuiSvgIcon-fontSizeInherit css-1vooibu-MuiSvgIcon-root"
                data-testid="StarIcon"
                focusable="false"
                viewBox="0 0 24 24"
              >
                <path
                  d="M12 17.27L18.18 21l-1.64-7.03L22 9.24l-7.19-.61L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21z"
                />
              </svg>
            </span>
          </span>
        </span>
      </div>
    </div>
    <p>
       test test test
    </p>
  </div>
</div>
`;
